README.md

```diff
@@ -4,690 +4,214 @@
-|`target`|`undefined`|Instead of using `node`, use `target` instead.  This is designed for supporting integration into other APIs and is not useful by itself.|
-|`thisArg`|`undefined`|By default, the resolved listener will be called with a `this` of `undefined`. Alternatively you can supply a `thisArg` to provide a different `this` when calling.|
-
-An example:
-
-```typescript
-const node = v('div', { key: 'root' }, [
-    w('sub-widget', { key: 'left', onClick: listener }, [ 'content' ]),
-    w('sub-widget', { key: 'right', onClick: listener }, [ 'content' ])
-]);
-
-callListener(node, 'onClick', {
-    args: [ event ],
-    key: 'left'
-});
-```
-
-### support/d
-
-In testing expected virtual DOM, it can be overly verbose to regenerate your virtual DOM every time you have changed the conditions
-that might effect the render.  Therefore the `support/d` module contains some helper functions which can be used to manipulate
-virtual DOM once it has been generated by the `v()` and `w()` virtual DOM functions.
-
-#### assignChildProperties()
-
-Shallowly assigns properties of a `WNode` or `HNode` indicated by an index.  The index can be a number, or it can be a string of numbers
-separated by commas to target a deeper child.  For example:
-
-```typescript
-const expected = v('div', [
-    v('ol', { type: 'I' }, [
-        v('li', { value: '3' }, [ 'foo' ]),
-        v('li', { }, [ 'bar' ]),
-        v('li', { }, [ 'baz' ])
-    ])
-]);
-
-assignChildProperties(expected, '0,2', { classes: css.highlight });
-```
-
-#### assignChildPropertiesByKey()
-
-Shallowly assigns properties of a `WNode` or `HNode` specified by its key. For example:
-
-```typescript
-const expected = v('div', [
-    v('ol', { type: 'I' }, [
-        v('li', { key: 'a', value: '3' }, [ 'foo' ]),
-        v('li', { }, [ 'bar' ]),
-        v('li', { }, [ 'baz' ])
-    ])
-]);
+const compareId = {
+	selector: '*', // all nodes
+	property: 'id',
+	comparator: (value: any) => typeof value === 'string' // checks the property value is a string
+};
 
-assignChildPropertiesByKey(expected, 'a', { classes: css.highlight });
+const h = harness(() => w(MyWidget, {}), [ compareId ]);
 ```
 
-#### assignProperties()
+For all assertions using the returned `harness` API will now only test identified `id` properties using the `comparator` instead of the standard equality.
 
-Shallowly assigns properties to a `WNode` or `HNode`.  For example:
+## selectors
 
-```typescript
-const expected = v('div', {
-    classes: css.root,
-    onclick: widget.listener
-}, [ 'content' ]);
-
-assignProperties(expected, {
-    classes: [ css.root, css.open ]
-});
-```
+The `harness` APIs commonly support a concept of css style selectors to target nodes within the virtual DOM for assertions and operations. The full list of supported selectors be found [here](https://github.com/fb55/css-select#supported-selectors).
```

Comment:
The full list of supported selectors be found [here](https://github.com/fb55/css-select#supported-selectors) -> Review the [full list of supported selectors](https://github.com/fb55/css-select#supported-selectors) for more information.